android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -108,16 +129,20 @@ protected void triggerLayoutInflation() {
             return;
         }
 
-        if (getIntent() != null) {
-            mChainId = getIntent().getStringExtra(Utils.CHAIN_ID);
-            mAssetSymbol = getIntent().getStringExtra(Utils.ASSET_SYMBOL);
-            mAssetName = getIntent().getStringExtra(Utils.ASSET_NAME);
-            mAssetId = getIntent().getStringExtra(Utils.ASSET_ID);
-            mContractAddress = getIntent().getStringExtra(Utils.ASSET_CONTRACT_ADDRESS);
-            mAssetLogo = getIntent().getStringExtra(Utils.ASSET_LOGO);
-            mAssetDecimals =
-                    getIntent().getIntExtra(Utils.ASSET_DECIMALS, Utils.ETH_DEFAULT_DECIMALS);
-            mCoinType = getIntent().getIntExtra(Utils.COIN_TYPE, CoinType.ETH);
+        final Intent intent = getIntent();
+        if (intent != null) {
+            mChainId = intent.getStringExtra(Utils.CHAIN_ID);
+            mAssetSymbol = intent.getStringExtra(Utils.ASSET_SYMBOL);
+            mAssetName = intent.getStringExtra(Utils.ASSET_NAME);
+            mAssetId = intent.getStringExtra(Utils.ASSET_ID);
+            mCoinMarket = intent.getBooleanExtra(Utils.COIN_MARKET, false);
+            mMarketCapRank = intent.getIntExtra(Utils.MARKET_CAP_RANK, -1);
+            mVolume24Hour = intent.getDoubleExtra(Utils.TOTAL_VOLUME, -1);
+            mMarketCap = intent.getDoubleExtra(Utils.MARKET_CAP, -1);
+            mContractAddress = intent.getStringExtra(Utils.ASSET_CONTRACT_ADDRESS);
+            mAssetLogo = intent.getStringExtra(Utils.ASSET_LOGO);
+            mAssetDecimals = intent.getIntExtra(Utils.ASSET_DECIMALS, Utils.ETH_DEFAULT_DECIMALS);
+            mCoinType = intent.getIntExtra(Utils.COIN_TYPE, CoinType.ETH);
```

Comment:
nit: we can now pass [serialised mojom object](https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/util/WalletUtils.java#L97-L107).

Reply:
Yes, I've considered that, but serialization and deserialization of an entire class is quite expensive. I would adopt that for a case where the fields are many. For a few doubles and two strings I preferred to stick to the fastest possible way.